product_variant_auto_creation_selectable/models/pricelist.py

```diff
@@ -0,0 +1,159 @@
+# -*- coding: utf-8 -*-
+# (c) 2015 Oihane Crucelaegui - AvanzOSC
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, fields, tools, api, exceptions, _
+
+
+class ProductPricelist(models.Model):
+    _inherit = 'product.pricelist'
+
+    @api.model
+    def _price_rule_get_multi(self, pricelist, products_by_qty_by_partner):
+        context = self.env.context.copy() or {}
```

Comment:
I think I found why is this required.

As an example, when working in sale with templates (no variant created) it does not know how much does it cost so it gets the template cost and the price extra from the attributes to provide a price unit.